runtime/src/bank.rs

```diff
@@ -7316,14 +7332,49 @@ impl Bank {
         new_feature_activations: &HashSet<Pubkey>,
     ) {
         for builtin in BUILTINS.iter() {
-            if let Some(feature_id) = builtin.feature_id {
-                let should_apply_action_for_feature_transition =
-                    if only_apply_transitions_for_new_features {
+            // The `builtin_disabled` flag is used to handle the case where a
+            // built-in is scheduled to be enabled by one feature gate and
+            // later migrated to Core BPF by another.
+            // There should never be a case where a built-in is set to be
+            // migrated to Core BPF and is also set to be enabled on feature
+            // activation on the same feature gate. However, the
+            // `builtin_disabled` flag will handle this case as well, electing
+            // to first attempt the migration to Core BPF.
+            // This will fail gracefully, and the built-in will subsequently be
+            // enabled, but it will never be migrated to Core BPF.
```

Comment:
It fails gracefully because `migrate_to_core_bpf(..)` is fallible and caught by `if let Err(e)` in the block below. Everything within that module returns a `Result<(), CoreBpfMigrationError>`, which bubbles all the way out to here intentionally.

I can definitely elaborate on what might get thrown here, though. It's all wrapped within the module.

Reply:
Yes, that's correct. Updated the comment.